doc/source/flexibility/compressed_columns.rst

```diff
@@ -0,0 +1,213 @@
+.. _compressed_columns:
+
+====================================
+Compressed columns with dictionaries
+====================================
+
+In :rn:`5.6.33-79.0` |Percona Server| has been extended with a new per-column
+compression feature. It is a data type modifier, independent from user-level
+SQL and |InnoDB| data compression, that causes the data stored in the column to
+be compressed on writing to storage and decompressed on read. For all other
+purposes the data type is identical to the one without modifier, i.e. no new
+data types are created.
+
+This feature provides:
+
+* a better compression ratio for text data which consist of a large number of
+  predefined words (e.g. JSON or XML) using compression methods with static
+  dictionaries,
+* in contrast to |InnoDB| row compression method it provides a way to select
+  which columns in the table must be compressed.
+
+Specifications
+==============
+
+The feature is limited to InnoDB/XtraDB storage engine and to:
+
+* ``BLOB`` (including ``TINYBLOB``, ``MEDIUMBLOB``, ``LONGBLOG``),
+
+* ``TEXT`` (including ``LONGTEXT``),
+
+* ``VARCHAR``, and
+
+* ``VARBINARY`` columns.
+
+The syntax to declare a compressed column is using an extension of an existing
+``COLUMN_FORMAT`` modifier: ``COLUMN_FORMAT COMPRESSED``. If this modifier is
+applied on an unsupported column type or storage engine, an error is returned.
+
+The compression can be specified:
+
+* at the table create time:
+  ``CREATE TABLE ... (..., foo BLOB COLUMN_FORMAT COMPRESSED, ...);``
+
+* or a table can be modified to compress/decompress a column later:
+  ``ALTER TABLE ... MODIFY [COLUMN] ... COLUMN_FORMAT COMPRESSED``, or
+  ``ALTER TABLE ... CHANGE [COLUMN] ... COLUMN_FORMAT COMPRESSED``.
+
+To decompress a column, a ``COLUMN_FORMAT`` value any other than ``COMPRESSED``
+can be specified: ``FIXED``, ``DYNAMIC``, or ``DEFAULT``. If there is a column
+compression/decompression request in an ``ALTER TABLE``, it is forced to the
+``COPY`` algorithm.
+
+Two new variables: :variable:`innodb_compressed_columns_zip_level` and
+:variable:`innodb_compressed_columns_threshold` have been implemented.
+
+Compression dictionary support
+==============================
+
+To achieve better compression ration on relatively small individual data items,
+it is possible to pre-define compression dictionary, which is a set of strings
+for each compressed column.
+
+The compression dictionary is stored in a new system |InnoDB| table or tables.
```

Comment:
Dictionary data itself is stored in one table - `SYS_ZIP_DICT`
`SYS_ZIP_DICT_COLS` is only used to store links between fields and dictionaries.